File: src/learning_communication/scripts/c_listener.py
#! /usr/bin/python
import rospy
from learning_communication.msg import Person

def myCallback(msg):
    rospy.loginfo('receive: {0}, {1}'.format(msg.name, msg.age))

def c_listener():
    rospy.init_node("c_listener", anonymous=True)
    rospy.Subscriber("c_talker", Person, queue_size=100, callback=myCallback)
    rospy.spin()

if __name__ == "__main__":
    try:
        c_listener()
    except rospy.ROSInterruptException:
        pass

File: src/learning_communication/src/c_listener.cpp
#include "ros/ros.h"
#include "learning_communication/Person.h"


void myCallback(const learning_communication::Person::ConstPtr &msg)
{
    ROS_INFO("get custom msg: %s, %d", msg->name.c_str(), msg->age);
}
int main(int argc, char *argv[])
{
    ros::init(argc, argv, "c_listener");
    ros::NodeHandle n;
    ros::Subscriber suber = n.subscribe("c_puber", 1000, myCallback);
    ros::spin();
    return 0;
}
